modules/core/src/main/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImpl.java

```diff
@@ -453,6 +458,22 @@ private void leaveBusy() {
     }
 
     /** {@inheritDoc} */
+    @Override public void timeout(long timeout) {
+        if (timeout < -1)
+            throw new IllegalArgumentException();
```

Comment:
Add description to "IllegalArgumentException" saying that this is a wrong timeout value.